packages/sui-mono/bin/sui-mono-release.js

```diff
@@ -142,9 +147,61 @@ const checkIsMasterBranchActive = async ({status, cwd}) => {
   }
 }
 
+const execute = async (cmd, full) => {
+  try {
+    console.log('--->', cmd)
+    const [resp] = await shell(cmd)
+    const output = full ? resp : resp.stdout
+    console.log(output)
+    return output
+  } catch (e) {
+    const output = full ? e : e.stderr
+    console.log(output)
+    return e
+  }
+}
+
+const automaticRelease = async ({
+  githubToken,
+  githubUser,
+  githubEmail,
+  cwd
+}) => {
+  const repoURL = await execute('git config --get remote.origin.url')
+  const gitURL = GitUrlParse(repoURL).toString('https')
+  const authURL = new URL(gitURL)
+  authURL.username = githubToken
+
+  await exec(`git pull --unshallow`, {cwd})
+  await exec(`git config --global user.email "${githubEmail}"`, {cwd})
+  await exec(`git config --global user.name "${githubUser}"`, {cwd})
+  await exec('git remote rm origin', {cwd})
+  await exec(`git remote add origin ${authURL} > /dev/null 2>&1`, {cwd})
+  await exec(`git checkout master`, {cwd})
+  await exec(`git pull origin master`, {cwd})
+}
+
+const isAutomaticRelease = ({githubToken, githubUser, githubEmail}) => {
+  return githubToken && githubUser && githubEmail
+}
+
 checker
   .check()
-  .then(status => checkIsMasterBranchActive({status, cwd: process.cwd()}))
+  .then(async status => {
+    isAutomaticRelease({
+      githubEmail: program.githubEmail,
+      githubToken: program.githubToken,
+      githubUser: program.githubUser
+    })
+      ? await automaticRelease({
+          githubEmail: program.githubEmail,
+          githubToken: program.githubToken,
+          githubUser: program.githubUser,
+          cwd: process.cwd()
+        })
+      : await checkIsMasterBranchActive({status, cwd: process.cwd()})
```

Comment:
Bit shorter and easier to read.

```suggestion
   const {githubEmail, githubToken, githubUser} = program
    isAutomaticRelease({
      githubEmail,
      githubToken,
      githubUser
    })
      ? await automaticRelease({
          githubEmail,
          githubToken,
          githubUser,
          cwd: process.cwd()
        })
      : await checkIsMasterBranchActive({status, cwd: process.cwd()})
```